plugins/UsersManager/TokenNotifications/TokenNotifierTask.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link    https://matomo.org
+ * @license https://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\UsersManager\TokenNotifications;
+
+use Exception;
+use Piwik\Container\StaticContainer;
+use Piwik\Date;
+use Piwik\Log\LoggerInterface;
+use Piwik\Option;
+use Piwik\Plugin\Manager as PluginManager;
+use Piwik\Scheduler\Schedule\Daily;
+use Piwik\Scheduler\Scheduler;
+use Piwik\Scheduler\Task;
+
+/**
+ * Send token notifications for each provider
+ */
+class TokenNotifierTask extends Task
```

Comment:
To have this task fully available we also need an entry in the `UsersManager\Tasks::schedule()` function.

Something like `$this->scheduleTask(new TokenNotifierTask());` should work.